orangecontrib/text/widgets/ownyt.py

```diff
@@ -10,14 +10,14 @@
 from orangecontrib.text.corpus import Corpus
 from orangecontrib.text.nyt import NYT, MIN_DATE
 from orangecontrib.text.widgets.utils import CheckListLayout, DatePickerInterval, QueryBox, \
-    gui_require, OWConcurrentWidget, asynchronous
+    gui_require, asynchronous
 
 
 class IO:
     CORPUS = "Corpus"
 
 
-class OWNYT(OWConcurrentWidget):
+class OWNYT(OWWidget):
```

Comment:
thank you so much, @nikicc 
I vote for an optional `should_raise` (or `strict`) argument in `callback` decorator. Then we can interpret `StopExcecution` error as trying to call a callback while the thread was stopped.